audb/core/publish.py

```diff
@@ -39,8 +39,8 @@ def job(table: str):
             deps.add_meta(file, table, checksum, version)
             tables.append(table)
         elif checksum != deps.checksum(file):
-            deps.data[file][define.DependField.CHECKSUM] = checksum
-            deps.data[file][define.DependField.VERSION] = version
+            deps.set_value(file, define.DependField.CHECKSUM, checksum)
+            deps.set_value(file, define.DependField.VERSION, version)
```

Comment:
We have a few unsymmetrical code lines, which of course doesn't make me happy :)

In the first `if`-statement we use ` deps.add_meta()` and in the checksum part we use `deps.set_value()`, but for adding media we use `dep.add_media()` in both cases.

Reply:
Yes, this asymmetry is not nice and we should get rid of it.

Reply:
I had a look at this, there seems to be no reason to use `deps.set_values()`. We could simply do:
```python 
        if file not in deps or checksum != deps.checksum(file):
            deps.add_meta(file, table, checksum, version)
            tables.append(table)
```
instead.

If have not commited those changes yet as it will result in conflicts. So I will wait until https://github.com/audeering/audb/pull/52 and related pull requests are finished.

Reply:
> I had a look at this, there seems to be no reason to use `deps.set_values()`.

Mhh, in case of `set_value()` we change the value of an existing entry, in case of `add_meta()` we add a new entry. I find calling `add_meta()` to change an entry a bit odd. So I would propose to either stay with two functions or find a better name for it.

Reply:
We do the same for `add_media()`, there we also call it to update an entry:

```python
        if file not in deps:
            checksum = audbackend.md5(path)
            if file in archives:
                archive = archives[file]
            else:
                archive = audeer.uid(from_string=file.replace('\\', '/'))
            deps.add_media(db_root, file, archive, checksum, version)
        elif not deps.is_removed(file):
            checksum = audbackend.md5(path)
            if checksum != deps.checksum(file):
                archive = deps.data[file][define.DependField.ARCHIVE]
                deps.add_media(db_root, file, archive, checksum, version)
```

I see the following solution:
* we use `add_media()` and `add_meta()` only for the first part in both cases and use `set_value()` in the other cases
* we use `add_media()` and `add_meta()` only for the first part in both cases and use `update_media()` and `update_meta()` in the second case which will replace `set_value()`

Reply:
Ok, then let's [kiss](https://en.wikipedia.org/wiki/KISS_principle) and stick to `add_media()` and `add_meta()` and add in the docstring that it either appends a new entry or updates if it already exists.

Reply:
OK, I'm using now `_add_media()` and `_add_meta()` in all cases.